docs/spec/concepts.rst

```diff
@@ -3,6 +3,336 @@
+not in the domain of the subtype, supertype, or equivalence relations on static
+types described above.
+
+To relate gradual types more generally, we define a **materialization**
+relation. Materialization transforms a "more dynamic" type to a "more static"
+type. Given a gradual type ``A``, if we replace zero or more occurrences of
+``Any`` in ``A`` with some gradual type (which can be different for each
+occurrence of ``Any``), the resulting gradual type ``B`` is a materialization
+of ``A``. (We can also materialize a :ref:`Callable` type by replacing ``...``
+with any type signature, and materialize ``tuple[Any, ...]`` by replacing it
+with a determinate-length tuple type.)
+
+For instance, ``tuple[int, str]`` (a fully static type) and ``tuple[Any, str]``
+(a gradual type) are both materializations of ``tuple[Any, Any]``. ``tuple[int,
+str]`` is also a materialization of ``tuple[Any, str]``.
+
+If ``B`` is a materialization of ``A``, we can say that ``B`` is a "more
+static" type than ``A``, and ``A`` is a "more dynamic" type than ``B``.
+
+The materialization relation is both transitive and reflexive, so it defines a
+preorder on gradual types.
+
+.. _`consistent`:
+
+Consistency
+-----------
+
+We define a **consistency** relation on gradual types, based on
+materialization.
+
+A fully static type ``A`` is consistent with another fully static type ``B`` if
+and only if they are the same type (``A`` is equivalent to ``B``).
+
+A gradual type ``A`` is consistent with a gradual type ``B``, and ``B`` is
+consistent with ``A``, if and only if there exists some fully static type ``C``
+which is a materialization of both ``A`` and ``B``.
+
+``Any`` is consistent with every type, and every type is consistent with
+``Any``. (This follows from the definitions of materialization and consistency
+but is worth stating explicitly.)
+
+The consistency relation is not transitive. ``tuple[int, int]`` is consistent
+with ``tuple[Any, int]``, and ``tuple[Any, int]`` is consistent with
+``tuple[str, int]``, but ``tuple[int, int]`` is not consistent with
+``tuple[str, int]``.
+
+The consistency relation is symmetric. If ``A`` is consistent with ``B``, ``B``
+is also consistent with ``A``. It is also reflexive: ``A`` is always consistent
+with ``A``.
+
+.. _`assignable`:
+
+The assignable-to (or consistent subtyping) relation
+----------------------------------------------------
+
+Given the materialization relation and the subtyping relation, we can define
+the **consistent subtype** relation over all types. A type ``B`` is a
+consistent subtype of a type ``A`` if there exists a materialization ``A'`` of
+``A`` and a materialization ``B'`` of ``B``, where ``A'`` and ``B'`` are both
+fully static types, and ``B'`` is a subtype of ``A'``.
+
+Consistent subtyping defines "assignability" for Python.  An expression can be
+assigned to a variable (including passed as a parameter or returned from a
```

Comment:
Nit: "passed as a parameter" should probably be "passed as an argument" or "passed to a parameter"